build/aten/src/ATen/Functions.h

```diff
@@ -2238,12 +2239,15 @@ static inline Tensor & baddbmm_out(Tensor & out, const Tensor & self, const Tens
 #endif
 }
 static inline Tensor bartlett_window(int64_t window_length, const TensorOptions & options) {
+    return bartlett_window(window_length, typeMetaToScalarType(options.dtype()), options.layout(), options.device(), options.pinned_memory());
+}
+static inline Tensor bartlett_window(int64_t window_length, c10::optional<ScalarType> dtype, c10::optional<Layout> layout, c10::optional<Device> device, c10::optional<bool> pin_memory) {
 #ifdef USE_STATIC_DISPATCH
-    return TypeDefault::bartlett_window(window_length, options);
+    return TypeDefault::bartlett_window(window_length, dtype, layout, device, pin_memory);
 #else
-    globalLegacyTypeDispatch().initForTensorTypeSet(at::detail::multi_dispatch_tensor_type_set(options));
+    //globalLegacyTypeDispatch().initForTensorTypeSet(options.type_set()); <<<-------- THIS HAS TO BE FIXED + call down 2 lines
```

Comment:
Will have to add some overload to accept something else but options

Reply:
What is wrong here?

Reply:
We dont have `options` object here anymore so we will have to extract/redesign/refactor the `type_set()` logic